pkg/k8sutil/metadata.go

```diff
@@ -35,6 +35,22 @@ func NodeAnnotationCondition(selector fields.Selector) watch.ConditionFunc {
 	}
 }
 
+// GetNodeRetry gets a node object, retrying up to DefaultBackoff number of times if it fails
+func GetNodeRetry(nc v1core.NodeInterface, node string) (*v1api.Node, error) {
+	var apiNode *v1api.Node
+	err := RetryOnError(DefaultBackoff, func() error {
+		n, getErr := nc.Get(node, v1meta.GetOptions{})
+		if getErr != nil {
+			return fmt.Errorf("failed to get node %q: %v", node, getErr)
+		}
+
+		apiNode = n
```

Comment:
nit: Seems like ``apiNode`` could be set where ``n`` is set. Not worth blocking over though.

Reply:
Hmm, OTOH doing it this way makes it easy to verify that `apiNode` can only ever be set if we're done retrying.